src/libs/antares/study/area/links.cpp

```diff
@@ -471,7 +548,20 @@ bool AreaLinksLoadFromFolder(Study& study, AreaList* l, Area* area, const AnyStr
         link.comments.clear();
         link.displayComments = true;
 
-        ret = link.loadTimeSeries(study, folder) && ret;
+        if (loadTSGen)
+        {
```

Comment:
About block : 
```C++
if (loadTSGen)
{
    ...
}
```
Too many implementation details we don't need to know about at this level.
We should call a function instead : 
```C++
if (loadTSGen)
    initializeTSgenerationPrepro(link);
```

Reply:
Furthermore, here we initialize the **prepro**, and in a further step (right after) we load its data matrix.
If we're handling with **prepro**, why not do in one step.
So for me this initialize step should be moved to the **loadTSgenerator** step (see remark above).
This would remove a **if (loadTSGen)** block.

Reply:
Yes, done